okhttp-testing-support/src/main/kotlin/okhttp3/DelegatingSSLSession.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import java.security.Principal
+import java.security.cert.Certificate
+import javax.net.ssl.SSLPeerUnverifiedException
+import javax.net.ssl.SSLSession
+import javax.net.ssl.SSLSessionContext
+import javax.security.cert.X509Certificate
+
+/** An [SSLSession] that delegates all calls.  */
+abstract class DelegatingSSLSession(protected val delegate: SSLSession?) : SSLSession {
```

Comment:
Why nullable? It’d be simpler to throw in the `get()` function if you wanna return null sometimes.

Reply:
(this is consistent with the original Java. Merging as-is; you can follow-up if you like!)